### PR TITLE
bears/java: Add InferBear

### PR DESCRIPTION
**Pull request by aachiritoaei**
Adds an InferBear for Java

Fixes https://github.com/coala-analyzer/coala-bears/issues/30

### CODE REVIEW

**Comment by Adrianzatreanu** [comment]
it seems you need a rebase.

**Comment by sils1297** [comment]
Hey, you might want to correct the authorship info to the commit so it'll be correctly attributed to you

**Comment by sils1297** [comment]
maybe @AbdealiJK or @Redridge or @Adrianzatreanu can help you get your tests passing

**Comment by tushar-rishav** [comment]
@aachiritoaei Hey! Seems like you need to first add opam PPA to the list of sources? Probably you need to add `ppa:avsm/ppa` before installing opam?:)

**Comment by aachiritoaei** [comment]
Yes, thank you for all the suggestions until now. I'm trying to fix all the errors given by this opam thing, seems pretty buggy. It's weird that I haven't had any of these on my machine.

**Comment by Makman2** [comment]
@aachiritoaei could you link the commit author to your github account? then it's easier to track whom a commit belongs to :+1:

File: circle.yml [hunk 1]
```diff
@@ -8,11 +8,14 @@ dependencies:
...
     - ~/dart-sdk/bin
     - ~/.cabal
+    - ~/infer-linux64-v0.7.0
   pre:
     - echo 'export PATH=$PATH:~/coala-bears/node_modules/.bin' >> ~/.circlerc
     - echo 'export LINTR_COMMENT_BOT=false' >> ~/.circlerc
     - echo 'export PATH=$PATH:~/dart-sdk/bin' >> ~/.circlerc
     - echo 'export PATH=$PATH:~/.cabal/bin' >> ~/.circlerc
+    - echo 'export PATH=$PATH:~/infer-linux64-v0.7.0/infer/bin' >> ~/.circlerc
+    - echo 'alias python2.7=python' >> ~/.circlerc
```
Comment (SanketDG) [inline circle.yml]: I am not sure what you want to do here, but if you want to use `python2.7`, you will have to set an exact local version explicitly.

Something like
````
pyenv install 2.7.10
pyenv local 2.7.10
```

https://github.com/yyuu/pyenv/issues/34

**Comment by Adrianzatreanu** [comment]
@aachiritoaei circleci failed due to having a command that requires input in there. its awaiting a yes/no response. therefore,add to the command ``-y``. should be smt like this ``$ pyenv install 2.7.10 -y``

**Comment by Adrianzatreanu** [comment]
same issue ;/ also, need a rebase; unack ffd8696

**Comment by sils1297** [comment]
hey you do need to do a rebase, there will be conflicts, I suggest you install `meld` and then run `git mergetool`

**Comment by aachiritoaei** [comment]
Had too many errors while solving conflicts. I'll open a new PR.

**Pull request closed without merging**